index.go

```diff
@@ -102,6 +102,18 @@ func (b *Batch) Reset() {
 	b.internal.Reset()
 }
 
+// Merge performs the merge of the internal Index.Batch
+// of the given batch to the self.
+func (b *Batch) Merge(other *Batch) {
```

Comment:
This method is unexpected.  Unfortunately, documents go through the mapping process when they are added to the Batch (initially).  With this method, I could create batches for 2 different indexes (which have difference mappings) then merge them, and add that batch to an index.  It's not wrong exactly, as you can do this directly with IndexAdvanced anyway, but it opens the door for more problems.

Reply:
I see, Does it make sense to take care of that conditional check inside this merge method?
like checking the index Mappings of the Batches for equality and merge only upon batches are of the same index Mapping()?